nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -206,7 +208,13 @@ class BtActionNode : public BT::ActionNodeBase
       // The following code corresponds to the "RUNNING" loop
       if (rclcpp::ok() && !goal_result_available_) {
         // user defined callback. May modify the value of "goal_updated_"
-        on_wait_for_result();
+        on_wait_for_result(feedback_);
+
+        // reset feedback to avoid stale information
+        {
+          std::lock_guard<std::mutex> lock_guard(feedback_mutex_);
```

Comment:
What's the work flow you're imagining? Users check if the `feedback_` is nullptr? That should be fine, but would be good at least in the signature of `getFeedback()` and `on_wait_for_result()` that it might be nullptr if no new feedback is received so that there's explicit documentation of that (which most people will miss/ignore, but at least we documented it)

Reply:
That's exactly it although I'm not seeing too much use of the `getFeedback()` function. `on_wait_for_result()` makes the most sense to give access to feedback to the user.

Another way that would make sense API wise would be to add an `on_feedback()` virtual method like the other `on_***` methods that can be overridden by the user to get access to the latest feedback only when it is available. This would remove the need to check for nullptr, which I agree would be missed by most users. What do you think?

Reply:
Agreed, the get feedback was an either/or kind of situation (change `on_wait` or add a `getFeedback()`) but I suppose it doesn't hurt to have both, but you could get rid of `getFeedback` and that would be prefectly fine to me